manifests/package.pp

```diff
@@ -22,17 +26,41 @@
     }
 
     'RedHat': {
+      $pkg_title = 'sensu'
+      $pkg_name = 'sensu'
+      $pkg_source = undef
+
       class { '::sensu::repo::yum': }
 
       $pkg_require = undef
     }
 
+    'windows': {
+      $repo_require = undef
+
+      $pkg_version = inline_template("<%= scope.lookupvar('sensu::version').sub(/(.*)\\./, '\\1-') %>")
+      $pkg_title = 'sensu'
+      $pkg_name = 'Sensu'
+      $pkg_source = "C:\\Windows\\Temp\\sensu-${pkg_version}.msi"
+      $pkg_require = "Archive[${pkg_source}]"
+
+      archive { $pkg_source:
```

Comment:
[puppet-archive](https://github.com/puppet-community/puppet-archive) is a puppet-community module used for, amongst other things, downloading files in a platform agnostic way. 

There are a number of other options and I am happy to alter this to whatever you prefer.